### DISCUSSION TITLE
Workbench Beta Phase 2022-10-24 to 2023-04-03

### DISCUSSION THREAD

**Discussion by zkamvar** [datacarpentry/r-socialsci#411] [Announcements]
Hello @datacarpentry/r-socialsci-maintainers,

Thank you for your patience during the last quarter. 
I am pleased to announce that the **transition to beta phase for this lesson will take place in 3 weeks on 2022-10-23**.

## Timing

the beta phase is divided into the following schedule of stages

 - 2022-10-23: pre-beta stage (two repos, two sites)
 - 2022-11-21: beta stage (one repo, two sites)
 - 2023-01-30: pre-release stage (one repo, one site) 

Please let me know if this does not work for you and I will do my best to accommodate. 
Importantly: **please resolve (close or merge) any open pull requests before 2022-11-21** because the beta stage will involve changing the folder structure of this repository which will render any existing pull requests invalid.

## Pre Beta Tasks

During this first pre-beta stage of the Beta Phase, I will give you access to https://github.com/fishtree-attempt/r-socialsci and encourage you to test out the following things on the lesson to get familiar with The Workbench. Here are some suggestions:

 - clone the lesson and build it locally
 - (optionally trigger and) merge and automated workflow update PRs that @carpentries-bot opens
 - review a PR that includes some failing accessibility checks
 - (R lessons only) review a PR using the preview of rendered changes
 - fix a simple typo in a lesson - perhaps in a relatively non-obvious location e.g. in the setup instructions
 - write a link from one episode to another in the same lesson 
 
I will be recording a short video describing the changes you will see in the next week, so be sure to keep your eyes out for that.

**Comment by eirini-zormpa** [comment]
Awesome, thank you for the update Zhian!

**Comment by zkamvar** [comment]
Hello @datacarpentry/r-socialsci-maintainers,

I have opened #423 and created the pre-beta snapshot of the lesson at https://github.com/fishtree-attempt/r-socialsci. I have invited all the maintainers on this lesson to a team that has maintain access for the lesson. This version of the lesson is a snapshot and is available for you to play around with and test out without risk of affecting the lesson as a whole. In 28 days, I will perform the full conversion on this lesson.

As a reminder, you can [check out the setup instructions](https://carpentries.github.io/sandpaper-docs/) to get yourself ready to build it locally.

Here are some tasks that you can try out during this 28 day period and please take a minute to give some feedback: https://carpentries.typeform.com/to/KRBl4IZM

 - clone the lesson and build it locally
 - merge one of those automated workflow update PRs that is opened by @carpentries-bot
 - review a PR that includes some failing accessibility checks
 - (R lessons only) review a PR using the preview of rendered changes
 - create an inline instructor note
 - fix a simple typo in a lesson - perhaps in a relatively non-obvious location e.g. in the setup 
instructions
 - fix some broken syntax in a lesson
 - write a link from one episode to another in the same lesson
 
If you have any questions, feel free to ask!

**Reply by zkamvar** [reply]
Note: I have pushed forward the dates by one week as I was late in submitting #423. I have notified the maintainers by email.

**Reply by zkamvar** [reply]
Hello @datacarpentry/r-socialsci-maintainers,

Congratulations on merging #423 and entering into the pre-beta stage! :tada: 

If you have have not yet, I would encourage you to clone the sandbox repository (https://github.com/fishtree-attempt/r-socialsci) to your machines so that you can work with the lesson locally. I have updated the source commit for the sandbox to be e8394f3a82f8771d479dc4cc23c306b01ff8202e, which includes #425. **If you have previously cloned the sandbox, please make sure you pull before you make any changes** as I made a force push to the repository to include those changes. Git should automatically update your local copy, but if that does not work, you can delete your local copy and re-clone.

Thank you all again for participating!

**Comment by zkamvar** [comment]
Hi @datacarpentry/r-socialsci-maintainers,

We are coming up on two weeks until we enter the beta stage of the Workbench beta phase, during which this repository will be irreversibly altered. 

**Please note that if you are not yet prepared for this change, we can delay the transition**

Just a reminder that before the transition, _all of the [current pull requests](https://github.com/datacarpentry/r-socialsci/pulls) must be resolved_. 

I also _highly_ recommend for you to play around with [the snapshot repository](https://github.com/fishtree-attempt/r-socialsci)! You all have full maintain access and are free to do whatever you like. Here are some options:

- Check out the [automated pull request to update packages](https://github.com/fishtree-attempt/r-socialsci/pull/1) on the snapshot repository. I've written some [documentation of what to expect](https://carpentries.github.io/sandpaper-docs/pull-request). 
- Trigger an [automated pull request to update workflows](https://github.com/fishtree-attempt/r-socialsci/actions/workflows/update-workflows.yaml) (these were recently updated in sandpaper 0.10.7)
- Use `sandpaper::validate_lesson()` locally to find and improve inaccessible elements of the lesson 
- Add an instructor note or two inline for the lesson from [the narrative instructor notes](https://preview.carpentries.org/r-socialsci/instructor/instructor-notes.html#narrative)

As always, please [provide feedback](https://carpentries.typeform.com/to/KRBl4IZM) about your experience with the workbench.

**Reply by eirini-zormpa** [reply]
Hi Zhian, I'm so sorry but I have not had time to work on this. For me, it'd be great if we could delay by a couple of weeks. @jessesadler is delaying okay by you?

**Reply by jessesadler** [reply]
I will have some time to play around with the new workbench site this coming week. I am also ok if we want to delay things for a bit. Maybe we should check back in at the end of next week.

**Reply by zkamvar** [reply]
Thank you both for your quick replies! The last thing I want is to put extra pressure on you, so I will go ahead and delay the transition to the beta stage until 2022-12-05.

**Comment by zkamvar** [comment]
## Workbench Beta Phase Timeframe Adjustment

> Note: this is identical to the message I sent directly via email

The beta phase has been in progress since the end of October, but **because of various changing circumstances, I will be pushing back the beta phase timeline according to the revised schedule below**, synchronising all of the beta phase repositories. With this new timeline, you will have more time to explore the repository snapshots, I will have time to provide more clear communications, and—because everyone will go through each stage synchronously—you will all be able to exchange tips and tricks together. 

Note that Toby Hodges, Vini Salazar, and I will be hosting weekly helpdesk sessions, whose times we will send to you soon. 

Here is the revised schedule, which you can also find [on The Workbench beta phase website](https://carpentries.github.io/workbench/beta-phase.html):

 - [2022-12-12 (pre-beta)](https://carpentries.github.io/workbench/beta-phase.html#pre-beta): (re-)release all beta phase lessons to the pre-beta stage, which will have a parallel snapshot at https://github.com/fishtree-attempt/ hosting the Workbench preview on https://preview.carpentries.org/. 
    - During this time, we encourage you to explore the repository. Here is a [list of tasks to get started](https://carpentries.github.io/workbench/beta-phase.html#pre-beta-tasks). Remember, This preview version is temporary, so you do not have to worry about making any mistakes here. 
    - Take the time to merge or resolve any open PRs. And yes, this is a big task and it is helpful to [remember how to say 'no'](https://opensource.guide/best-practices/#learning-to-say-no) in a way that respects the contributor’s time and effort. If you feel like you need help with a particular PR, please contact Zhian and he will help craft a response.
 - [2023-02-06 (beta)](https://carpentries.github.io/workbench/beta-phase.html#beta): do in-place conversion of all beta phase lessons to use The Workbench by default. The default lesson site will still point to a snapshot of the jekyll version from 2023-02-06, and the preview lesson site will show the Workbench hosted on the main branch of the repository. 
    - IMPORTANT: before this happens, all maintainers MUST be aware of this change and must merge or resolve any open PRs. Remember that it’s okay to say no and/or contact Zhian for wordsmithing.
    - Zhian will create a video describing what this process is doing
    - The broader maintainer community will be alerted that they have three months to merge or resolve their open PRs at this time.
 - [2023-04-03 (pre-release)](https://carpentries.github.io/workbench/beta-phase.html#pre-release): The Workbench will become the default version of the lesson and the Jekyll version will be removed. A blog post will go out alerting everyone of the change. 


**We will make the conversion of all of our lessons to use The Workbench on 2023-05-01**


Thank you once again for agreeing to participate in the beta phase of The Carpentries Workbench. Your efforts in testing the infrastructure on your established lessons are invaluable to improving the workbench and I want to make sure that we take the time to get that right.

**Reply by jessesadler** [reply]
I just merged the PR to update the dates for the beta phase. The changes all sound good to me.

**Comment by zkamvar** [comment]
Hello and happy new year! 

This is a reminder that on **06 February 2023 (just under four weeks away), your lesson repositories will be transformed to use The Carpentries Workbench.** This is the irreversible [second stage of the beta phase](https://carpentries.github.io/workbench/beta-phase.html#beta) where the Workbench version of the lesson will be live and the styles version will be frozen.

### Before 06 February 2023

1. **(MUST DO)** Merge or resolve any open PRs (If you are unsure of how to resolve some PRs that need more time, [here's a guide on how to say 'no'](https://opensource.guide/best-practices/#learning-to-say-no))
2. **(recommended)** Try out the [pre-beta tasks to get familiar with The Workbench](https://carpentries.github.io/workbench/beta-phase.html#pre-beta)

### After 06 February 2023

The lesson repository will change with a trimmed-down git history that reflects authorship. **To prevent accidental overwriting of repository content, I will temporarily change your access to the repository as ‘read-only’.** To regain ‘maintain’ access,

1. **(MUST DO)** [make a fresh clone of or update](https://carpentries.github.io/workbench/beta-phase.html#beta-tasks) your lesson on your computer.
2. **(MUST DO)** respond here with `I am ready for write access :rocket:`

### Have Questions?

If you have any questions, would like help resolving PRs, or anything at all, you can email, slack, or schedule a meeting with me: https://calendly.com/zkamvar.

**Comment by zkamvar** [comment]
Hello Maintainers,

Today is the day the lesson transitions into the second stage of the beta phase :tada:

I have created lesson release for this lesson and also have downloaded an archive of the repository and all associated comments. Before the end of the day, I will do the following:

1. temporarily restrict push access for the maintainers while I make the transition
2. set the config to display the beta stage banner (red), which will indicate that the styles version is locked
3. perform the transition (here is a rough 3 and a half minute video of what you can expect after the transition: https://youtu.be/kvmDlgjV8NA)

Once the transition happens, I will give access to those who follow the instructions:

> * **(MUST DO)** [make a fresh clone of or update](https://carpentries.github.io/workbench/beta-phase.html#beta-tasks) your lesson on your computer.
> * **(MUST DO)** respond here with `I am ready for write access :rocket:`

Thank you all for the commitment to testing this out and for all your feedback so far!

**Comment by zkamvar** [comment]
The transition is done. When you have created a fresh clone of the lesson, please comment here with `I am ready for write access :rocket:` and I will grant you write access to the repository (note that we are going to leave the main branch protected).

**Comment by eirini-zormpa** [comment]
I am ready for write access 🚀

**Reply by zkamvar** [reply]
done!

**Comment by juanfung** [comment]
I am ready for write access :rocket:

**Reply by zkamvar** [reply]
done!

**Comment by jessesadler** [comment]
I am ready for write access :rocket:

**Reply by zkamvar** [reply]
Done!

**Comment by zkamvar** [comment]
Thank you all for your hard work getting the pull requests merged and giving feedback. 

Now that you all have access, here are some resources that can help you understand how to handle the transition:

1. What you can expect from the transition 📹: https://carpentries.github.io/workbench/beta-phase.html#beta
1. How to update your local clone 💻: https://carpentries.github.io/workbench/beta-phase.html#updating-clone
1. How to update (delete) your fork (if you have one) 📹: https://carpentries.github.io/workbench/faq.html#update-fork-from-styles

I will be creating an pull request from an out-of-date repository tomorrow so that you can see what happens. If everything works the way I've planned, the pull request should fail.

**Comment by zkamvar** [comment]
I have created a video that describes situations where folks will accidentally create PRs from styles repositories, as documented in [The Transition from Carpentries to Styles section in the Pull request documentation](https://carpentries.github.io/sandpaper-docs/pull-request.html#transition-from-carpentriesstyles): https://youtu.be/FRZKbCz-9Eo (note that this only applies to R-based lessons)

Some of you may have seen https://github.com/datacarpentry/r-socialsci/pull/433 come through and that was what I was demonstrating in the video.
